modules/dot-prefix-validation/src/main/java/org/elasticsearch/validation/DotPrefixValidator.java

```diff
@@ -93,11 +95,13 @@ public abstract class DotPrefixValidator<RequestType> implements MappedActionFil
 
     private final ThreadContext threadContext;
     private final boolean isEnabled;
+    private final boolean isStateless;
     private volatile Set<Pattern> ignoredIndexPatterns;
 
     public DotPrefixValidator(ThreadContext threadContext, ClusterService clusterService) {
         this.threadContext = threadContext;
         this.isEnabled = VALIDATE_DOT_PREFIXES.get(clusterService.getSettings());
+        this.isStateless = DiscoveryNode.isStateless(clusterService.getSettings());
```

Comment:
nit I had forgotten we already had this function to judge whether we're in stateless mode. cc @mosche to be aware.